### PR TITLE
Fail satisfy when resource limits exceed

### PR DESCRIPTION
**Pull request by sanket1729**
Satisfy can fail if the resource limits are exceeded

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ honggfuzz_fuzz = ["honggfuzz"]
 honggfuzz = { version = "0.5", optional = true }
 afl = { version = "0.8", optional = true }
 regex = { version = "1.4"}
-miniscript = { path = "..", features = ["fuzztarget", "compiler"] }
```
Comment (apoelstra) [inline fuzz/Cargo.toml:18]: Why are we changing this? The fuzzer won't be able to come up with hash preimages or signatures without the fuzztarget on.

Comment (sanket1729) [reply fuzz/Cargo.toml:18]: What does the `fuzztarget` feature do in rust-bitcoin? I think the only use we have in rust-miniscript is pass through for `rust-bitcoin`.

Comment (sanket1729) [reply fuzz/Cargo.toml:18]: I think I disabled them because I was not able to come up with secp Signature required for satisfier.

**Review by apoelstra — APPROVED** [review]
ack 70b987fa12f9c637f2688b1e96ac30353f7f6298

File: src/miniscript/context.rs [hunk 2]
```diff
@@ -294,6 +318,17 @@ impl ScriptContext for Segwitv0 {
         Ok(())
     }
 
+    fn check_witness<Pk: MiniscriptKey, Ctx: ScriptContext>(
+        witness: &[Vec<u8>],
+    ) -> Result<(), ScriptContextError> {
+        if witness_size(witness) > MAX_STANDARD_P2WSH_SCRIPT_SIZE {
+            return Err(ScriptContextError::MaxScriptSigSizeExceeded);
```
Comment (darosior) [inline src/miniscript/context.rs:325]: Hmm shouldn't it return a `MaxWitnessSizeExceeded` Error instead ?

**Pull request merged**